ComponentKit/Core/CKComponentAnimation.h

```diff
@@ -18,15 +18,27 @@
 struct CKComponentAnimation {
 
   /**
-   Creates a CKComponentAnimation that applies a CAAnimation to a CKComponent.
-
+   Creates a CKComponentAnimation that applies a CAAnimation to a CKComponent on the layer found at layerPath.
+   
    @note The CKComponent must create a UIView via its view configuration; or, it must be a CKCompositeComponent that
-   renders to a component that creates a view.
-
+   renders to a component that creates a view. The view must have a CALayer at the keypath that corresponds to 
+   layerPath.
+   
+   @example Suppose the component has a CAShapeLayer as the mask of the view's layer. The path of the mask could be animated
+   with: {myComponent, [CABasicAnimation animationWithKeypath:@"path"], "layer.mask"}
+   */
+  CKComponentAnimation(CKComponent* component, CAAnimation* animation, NSString* layerPath);
+  
+  /**
+   Creates a CKComponentAnimation that applies a CAAnimation to a CKComponent on the layer of the view.
+   
    @example {myComponent, [CABasicAnimation animationWithKeypath:@"position"]}
+   
+   @see CKComponentAnimation(CKComponent* component, CAAnimation* animation, NSString* layerPath)
    */
   CKComponentAnimation(CKComponent *component, CAAnimation *animation);
```

Comment:
http://componentkit.org/docs/avoid-overrides.html

just define a single function with a default value of nil for the layerPath parameter